kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTlsIT.java

```diff
@@ -0,0 +1,188 @@
+        vaultKmsMap.put("tlsInsecure", getTlsVaultKms(insecureTls()));
+    }
+
+    private static void startVault() {
+        vaultContainer = new VaultContainer<>(HASHICORP_VAULT)
+                .withVaultToken(VAULT_TOKEN)
+                .withEnv("VAULT_FORMAT", "json")
+                // in vault server -dev mode the 8200 port cannot be reconfigured to TLS and is also useful so that the
+                // internal vault commands we make while execed into the container can communicate with vault over plain
+                // without having to trust the TLS cert within the container. We do not expose the plain listener to
+                // prevent false positives if the test connected to the wrong endpoint.
+                .withExposedPorts(TLS_PORT)
+                .withCopyFileToContainer(MountableFile.forClasspathResource("Vault.hcl"), "/vault/config/Vault.hcl")
+                .withCopyFileToContainer(MountableFile.forHostPath(keys.serverCertPem()), "/vault/config/cert.pem")
+                .withCopyFileToContainer(MountableFile.forHostPath(keys.serverPrivateKeyPem()), "/vault/config/key.pem")
+                .withInitCommand(
+                        "secrets enable transit");
+        vaultContainer.setWaitStrategy(Wait.forHttp("/v1/sys/health").forPort(TLS_PORT).usingTls().allowInsecure().forStatusCode(200));
+        vaultContainer.start();
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        vaultContainer.stop();
+    }
+
+    @Test
+    void tlsConnectionFailsWithoutClientTrust() {
+        // passing null tls means we do not attempt to obtain a custom SSLContext from FilterFactoryContext
+        var tlsConfig = vaultConfig(null);
+        var keyName = "mykey";
+        createKek(keyName);
+        VaultKms service = new VaultKmsService().buildKms(tlsConfig);
+        var resolved = service.resolveAlias(keyName);
+        assertThat(resolved)
+                .failsWithin(Duration.ofSeconds(5))
+                .withThrowableThat().havingCause()
+                .isInstanceOf(SSLHandshakeException.class)
+                .withMessageContaining("unable to find valid certification path to requested target");
+    }
+
+    @ParameterizedTest
+    @MethodSource("vaultKms")
+    void testArbitraryKmsOperationSucceedsWithTls(String kms) {
+        VaultKms service = vaultKmsMap.get(kms);
+        var keyName = "mykey";
+        createKek(keyName);
+        var resolved = service.resolveAlias(keyName);
+        assertThat(resolved)
+                .succeedsWithin(Duration.ofSeconds(5))
+                .isEqualTo(keyName);
+    }
+
+    private ReadKeyData createKek(String keyId) {
+        return runVaultCommand(new TypeReference<>() {
+        }, "vault", "write", "-f", "transit/keys/%s".formatted(keyId));
+    }
+
+    private <D> D runVaultCommand(TypeReference<VaultResponse<D>> valueTypeRef, String... args) {
```

Comment:
we should extract the common methods.

Reply:
Created a TestVault class that hides the differences between TLS/non-TLS, unifies constructing the container and offers `runVaultCommand` along with methods to obtain the token and endpoint.